### PR TITLE
use new docker files

### PR DESCRIPTION
**Pull request by rndmh3ro**
This uses my own docker images (https://github.com/rndmh3ro/docker-ansible). These are just the base images with pre-installed Docker to save time and hassle when testing the roles.
This also enables full travis testing of all supported operating systems!

This too removes ansible 1.9 support, to better test the role. Also there's no more support for Ansible < 2.0, see https://groups.google.com/forum/#!topic/ansible-devel/6-6FdxZ94kc.

This also updates .kitchen.vagrant.yml same way as in PR https://github.com/dev-sec/ansible-os-hardening/pull/113.

### CODE REVIEW

File: .kitchen.vagrant.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  roles_path: ../ansible-mysql-hardening/
+  playbook: default.yml
+  requirements_path: requirements.yml
+  sudo_command: 'sudo -E -H'
+
+transport:
+  max_ssh_sessions: 5
+
+platforms:
+- name: ubuntu-12.04
```
Comment (shortdudey123) [inline .kitchen.vagrant.yml]: EOL'd

File: .kitchen.vagrant.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  driver_config:
+    box: opscode-ubuntu-12.04
+    box_url: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_ubuntu-12.04_chef-provisionerless.box
+- name: ubuntu-14.04
+  driver_config:
```
Comment (shortdudey123) [inline .kitchen.vagrant.yml]: `driver_config` is not needed since it defaults to bento boxes

File: .kitchen.yml [hunk 3]
```diff
@@ -1,62 +1,77 @@
...
-  driver_config:
-    box: debian-7
-    box_url: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_debian-7.8_chef-provisionerless.box
-- name: debian-8
-  driver_config:
-    box: debian-8
-    box_url: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_debian-8.1_chef-provisionerless.box
+- name: centos6-ansible-latest
+  driver:
+    image: rndmh3ro/docker-centos6-ansible:latest
```
Comment (shortdudey123) [inline .kitchen.yml]: docker stuff should be in `.kitchen.docker.yml`

File: .kitchen.yml [hunk 4]
```diff
@@ -1,62 +1,77 @@
 ---
 driver:
-  name: vagrant
+  name: docker
```
Comment (shortdudey123) [inline .kitchen.yml]: any reason to not use `dokken`?

File: .kitchen.vagrant.yml (hunk 1 above)
Comment (rndmh3ro) [inline .kitchen.vagrant.yml]: Removed.

File: .kitchen.vagrant.yml (hunk 2 above)
Comment (rndmh3ro) [inline .kitchen.vagrant.yml]: Removed.

File: .kitchen.yml (hunk 3 above)
Comment (rndmh3ro) [inline .kitchen.yml]: Docker is the default testing method in this role. What else should be there then?

File: .kitchen.yml (hunk 4 above)
Comment (rndmh3ro) [inline .kitchen.yml]: Ẁhat's the difference and advantages?

Comment (rndmh3ro) [inline .kitchen.yml]: I took a look at https://github.com/someara/kitchen-dokken.
It sounds good, but right now we use kitchen-docker in all our ansible-roles. So I'l write it on my todo-list to check out dokken.

File: .kitchen.yml (hunk 3 above)
Comment (shortdudey123) [inline .kitchen.yml]: hmm ok, looks like the other repos are non-standard as well so ignore this commend then :)

Comment (rndmh3ro) [inline .kitchen.yml]: Actually I'll change this to the bento boxes. I just discovered that bento has all of them.

File: .kitchen.yml (hunk 4 above)
Comment (shortdudey123) [inline .kitchen.yml]: Take a look at https://github.com/dev-sec/chef-mysql-hardening/blob/master/.kitchen.yml
tl;dr dokken abstracts away from of the docker stuff that you don't care about in a CI environment (i.e. the `run_opts` and `init`options you have in the travis config)

Your config looks like it works fine the way it is, so just a thought if you have time or for the future :)

**Comment by rndmh3ro** [comment]
Travis green, does someone have any objections against merging?

**Comment by artem-sidorenko** [comment]
LGTM
